processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -1345,6 +1364,20 @@ public boolean hasAccessibleConstructor() {
         return hasAccessibleConstructor;
     }
 
+    public boolean hasAccessibleDefaultConstructor() {
+        if ( hasAccessibleDefaultConstructor == null ) {
+            hasAccessibleDefaultConstructor = false;
+            List<ExecutableElement> constructors = ElementFilter.constructorsIn( typeElement.getEnclosedElements() );
+            for ( ExecutableElement constructor : constructors ) {
+                if ( constructor.isDefault() && !constructor.getModifiers().contains( Modifier.PRIVATE ) ) {
+                    hasAccessibleDefaultConstructor = true;
+                    break;
+                }
+            }
+        }
+        return hasAccessibleDefaultConstructor;
+    }
+
```

Comment:
This is related to an existing strange behaviour with `CommonMacros#initTargetObject` https://github.com/mapstruct/mapstruct/blob/c9b551aebc0c5b1d9b84936100e4d36e87bb32bb/processor/src/main/resources/org/mapstruct/ap/internal/model/macro/CommonMacros.ftl#L173

In that code, it is blindly calling `new X()` even if `X` does not have a default constructor. This is even called out in [NullValuePropertyMappingStrategy](https://github.com/mapstruct/mapstruct/blob/0e2832484032a1f94e9238ef3397586010b83e81/core/src/main/java/org/mapstruct/NullValuePropertyMappingStrategy.java#L48), but in this case since this is a nested object, we cannot work around it with a `defaultValue()`.

Thus this method was added so that `null` can be returned when a "default value" [cannot be constructed](https://github.com/ro0sterjam/mapstruct/blob/9e969e7308a49ef182ca82b86d72a896cf881e1a/processor/src/main/resources/org/mapstruct/ap/internal/model/InitDefaultValue.ftl#L27).

I'm considering just removing this logic and keeping it simple (i.e. let the user fix this issue by writing a custom mapper or create a default constructor).